fe/fe-core/src/main/java/org/apache/doris/nereids/PlannerAdapter.java

```diff
@@ -0,0 +1,60 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids;
+
+import org.apache.doris.analysis.Analyzer;
+import org.apache.doris.analysis.StatementBase;
+import org.apache.doris.common.UserException;
+import org.apache.doris.nereids.properties.PhysicalProperties;
+import org.apache.doris.nereids.trees.plans.PhysicalPlanTranslator;
+import org.apache.doris.nereids.trees.plans.PlanContext;
+import org.apache.doris.nereids.trees.plans.logical.LogicalPlanAdapter;
+import org.apache.doris.nereids.trees.plans.physical.PhysicalPlan;
+import org.apache.doris.planner.Planner;
+import org.apache.doris.qe.ConnectContext;
+import org.apache.doris.thrift.TQueryOptions;
+
+import java.util.ArrayList;
+
+/**
+ * This class is used for the compatibility and code reuse in.
+ * @see org.apache.doris.qe.StmtExecutor
+ */
+public class PlannerAdapter extends Planner {
```

Comment:
Whether to abstract the original planner out of an interface for compatibility between the old and new optimizers. Let both the old and new optimizer planner inherit from this interface. One is NereidsPlanner and the other is OriginPlanner.
Each implements its own plan function, and the return value is List<PlanFragment> .

At present, although this adaptor mode can be implemented, the abstraction is not good.

Reply:
Okay, I'll refactor it in the future